provider/core.rb

```diff
@@ -28,6 +28,112 @@ module Provider
     quiet: false
   }.freeze
 
+  # Responsible for generating a file
+  # with a given set of parameters.
+  class FileTemplate
```

Comment:
Does it make sense for this class to only have the general object stuff and then each provider can subclass it and add their own provider specific variables?

Reply:
That's definitely something we could do. It might be overly complicated though?

Reply:
I don't think it would be overly complicated, but it's something we can do in the future.

In my opinion this FileTemplate structure helps to show what fields are available from the core MM level, but it doesn't help for provider specific fields. At least in inspec some fields are available only when compiling certain templates, so looking at this file wouldn't tell you which fields are available during a given template compilation.